src/_pytest/warning_types.py

```diff
@@ -9,6 +9,14 @@ class PytestWarning(UserWarning):
     """
 
 
+class UnknownMarkWarning(PytestWarning):
```

Comment:
About creating a new warning, see my thoughts on the bottom of https://github.com/pytest-dev/pytest/pull/4830#discussion_r265801473.

cc @blueyed 

If we go down that route, we should introduce warning types for the other `PytestWarning` types we produce (I think there are just a few).

Reply:
> If we go down that route, we should introduce warning types for the other PytestWarning types we produce (I think there are just a few).

Just to be clear: if we decide to add more warning types, we should do it in a separate PR so as not to burden this one.

Reply:
OK, taking it out of this one.